workspace/workspace_test.go

```diff
@@ -153,3 +153,117 @@ func TestLocate(t *testing.T) {
 		assert.Equal(t, test.out, dirs, test.desc)
 	}
 }
+
+func TestSolutionPath(t *testing.T) {
+	root := filepath.Join("..", "fixtures", "solution-path", "creatures")
+	ws := New(root)
+
+	// An existing exercise.
+	path, err := ws.SolutionPath("gazelle", "ccc")
+	assert.NoError(t, err)
+	assert.Equal(t, filepath.Join(root, "gazelle-3"), path)
+
+	path, err = ws.SolutionPath("gazelle", "abc")
+	assert.NoError(t, err)
+	assert.Equal(t, filepath.Join(root, "gazelle-4"), path)
+
+	// A new exercise.
+	path, err = ws.SolutionPath("lizard", "abc")
+	assert.NoError(t, err)
+	assert.Equal(t, filepath.Join(root, "lizard"), path)
+}
+
+func TestPathContainsSolution(t *testing.T) {
+	root := filepath.Join("..", "fixtures", "path-contains-solution")
+
+	ok, err := PathContainsSolution(filepath.Join(root, "yepp"), "abc")
+	assert.NoError(t, err)
+	assert.True(t, ok)
+
+	ok, err = PathContainsSolution(filepath.Join(root, "nope"), "abc")
+	assert.NoError(t, err)
+	assert.False(t, ok)
+
+	_, err = PathContainsSolution(filepath.Join(root, "broken"), "abc")
+	assert.Error(t, err)
+}
```

Comment:
I added a test for this because I wanted to be sure that I was handling the difference between a NotExists error (which is not an error in this context) and a broken solution metadata file.